src/briefcase/commands/new.py

```diff
@@ -113,7 +127,10 @@ def validate_app_name(self, candidate):
                 "name, move to a different parent directory, or delete the "
                 "existing folder.".format(candidate=candidate)
             )
-
+        if iskeyword(candidate) or candidate in javareserved:
+            raise ValueError(
+                "App name may not contain Java or Python reserved words."
```

Comment:
As above - we should avoid calling out specific languages, and just refer to reserved words. For bonus points, we should also clarify what the reserved word *is* - i.e, say `The app name "switch" cannot be used, as it is a reserved word`.